examples/AzureFunction/Startup.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="Startup.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+[assembly: FunctionsStartup(typeof(Examples.AzureFunction.Startup))]
+
+namespace Examples.AzureFunction
+{
+    public class Startup : FunctionsStartup
+    {
+        private static TracerProvider tracerProvider;
+
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            // TODO: It does not appear that the OpenTelemetry.Extensions.Hosting package is compatible with Azure Functions.
```

Comment:
yea. its a known limitation in Function.  We can try to change the usage of our hosted service to something else. Perhaps `IStartupFilter` as used here : https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Implementation/ApplicationInsightsStartupFilter.cs#L38